### ISSUE TITLE
Foreground app loses focus after input source switch (Ubuntu 24.04, nVidia drivers)

### ISSUE THREAD

**Issue by serge-vk** [galets/gnome-keyboard-reset#9]
After the keyboard input was reset and I unlocked the session, the foreground app window has focus and I can start typing normally. However, If I try to switch the keyboard input to some other language without first bringing another window to the foreground, the current window loses focus. It looks as though there were a hidden window, which gets input focus, because no other open window is activated. This is quite annoying, because every single time after unlocking my session I have to cycle to a different window and back in order to restore input focus. It is even more annoying if I do not have to change input source from the default right away, because I forget about this issue and when I do switch it, there is a moment of disorientation before I realise that I cannot type until I cycle to a different window, then back.

I noticed this behaviour with the previous version 5 and it is the same with the current version 6.

**Comment by galets** [comment]
Need more context. Please provide: - what is initial setup; - what you do; - what is happening that's wrong

**Comment by serge-vk** [comment]
Thank you for responding. I have Ubuntu 24.04 with the default desktop, Extension Manager for Gnome with a few extensions (Astra Monitor, Custom accent colours, Lock Keys, OpenWeather Refined). I also have a bunch of applications installed, but that seems of no consequence. Steps to reproduce:

1. Install at least two input sources
2. Open an application window with text input (e.g. Text Editor) and switch to a non-default input source. You can type normally in the non-default input source.
3. Lock the desktop.
4. Unlock the desktop. Input source in the originally open application is now reset to default. You can type normally in it.
5. Try to switch the input source to the non-default. You should be able to continue typing normally, but instead, the window loses focus. When you press keys, nothing happens. 
6. Bring any other visible window to foreground and then go back to the original application window. Note that its input source is still default.
7. Try switching again to the non-default input source. This time it works and you can continue typing normally.

The wrong thing is that I should not have to switch out of the foreground window and back in order to select a different input source.

**Comment by galets** [comment]
I tried reproducing the problem using steps you described, and failed. At step 5, I can type into window normally, using any keyboard layout.

How did you identify this is gnome-keyboard-reset extension that is causing the issue? Does the problem go away when you disable extension?

**Comment by serge-vk** [comment]
I thought I verified it, but when I tried disabling this extension once again now to double-check, I still observe this problem. So, you are probably right and there is something else going on. I will try to disable everything that I have and check again. By the way, I also noticed that on a fresh installation of Ubuntu 24.04 I am unable to select the input source with the mouse (i.e. by clicking on the indicator and then on the desired language). It never worked, although there is no problem switching with the keyboard.
